src/Plugin/Commerce/ShippingMethod/CanadaPost.php

```diff
@@ -171,10 +177,18 @@ public function submitConfigurationForm(array &$form, FormStateInterface $form_s
   public function calculateRates(ShipmentInterface $shipment) {
     $rates = [];
 
-    // Only attempt to collect rates if an address exits on the shipment.
+    // Only attempt to collect rates if an address exists on the shipment.
     if (!$shipment->getShippingProfile()->get('address')->isEmpty()) {
-      $this->canadapost_rate_service->setShipment($shipment);
-      $rates = $this->canadapost_rate_service->getRates();
+      $order_id = $shipment->order_id->target_id;
+      $order = Order::load($order_id);
+      /** @var \Drupal\commerce_store\Entity\Store $store */
+      $store = $order->getStore();
+      $address = $store->getAddress();
```

Comment:
The store's address might be very different than the address from which parcels are dispatched. The store's address is the business address for legal or contact purposes. While the fulfillment center in most cases would be located elsewhere or even provided by a 3rd party service.
Since we only need the postcode for calculating shipping rates, we can simply add a mandatory configuration option on the shipping method.